### PR TITLE
allow to pass a new K to distort and undistort points as in opencv

### PR DESCRIPTION
**Pull request by lferraz**
#### Changes
add the new K as in opencv. 
https://vovkos.github.io/doxyrest-showcase/opencv/sphinx_rtd_theme/group_imgproc_transform.html#doxid-da-d54-group-imgproc-transform-1ga55c716492470bfe86b0ee9bf3a1f0f7e

Fixes # (issue)

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/geometry/calibration/distort.py [hunk 1]
```diff
@@ -96,9 +105,14 @@ def distort_points(points: torch.Tensor, K: torch.Tensor, dist: torch.Tensor) ->
     cy: torch.Tensor = K[..., 1:2, 2]  # princial point in y (Bx1)
     fx: torch.Tensor = K[..., 0:1, 0]  # focal in x (Bx1)
     fy: torch.Tensor = K[..., 1:2, 1]  # focal in y (Bx1)
+    new_cx: torch.Tensor = new_K[..., 0:1, 2]  # princial point in x (Bx1)
```
Comment (edgarriba) [inline kornia/geometry/calibration/distort.py:108]: cx is never used then ?

File: kornia/geometry/calibration/undistort.py [hunk 2]
```diff
@@ -54,6 +64,10 @@ def undistort_points(points: torch.Tensor, K: torch.Tensor, dist: torch.Tensor)
     cy: torch.Tensor = K[..., 1:2, 2]  # princial point in y (Bx1)
     fx: torch.Tensor = K[..., 0:1, 0]  # focal in x (Bx1)
     fy: torch.Tensor = K[..., 1:2, 1]  # focal in y (Bx1)
+    new_cx: torch.Tensor = new_K[..., 0:1, 2]  # princial point in x (Bx1)
```
Comment (edgarriba) [inline kornia/geometry/calibration/undistort.py:67]: Same as before with respect to new k

File: kornia/geometry/calibration/undistort.py [hunk 3]
```diff
@@ -21,7 +24,8 @@ def undistort_points(points: torch.Tensor, K: torch.Tensor, dist: torch.Tensor)
         dist: Distortion coefficients
             :math:`(k_1,k_2,p_1,p_2[,k_3[,k_4,k_5,k_6[,s_1,s_2,s_3,s_4[,\tau_x,\tau_y]]]])`. This is
             a vector with 4, 5, 8, 12 or 14 elements with shape :math:`(*, n)`.
-
+        new_K: New intrinsic camera matrix with shape :math:`(*, 3, 3)`. Default: None, in this case K is used.
+        iters: Number of undistortion iterations. Default: 5.
```
Comment (edgarriba) [inline kornia/geometry/calibration/undistort.py:28]:
```suggestion
        num_iters: Number of undistortion iterations. Default: 5.
```
Same for the other

File: kornia/geometry/calibration/distort.py [hunk 4]
```diff
@@ -67,6 +70,7 @@ def distort_points(points: torch.Tensor, K: torch.Tensor, dist: torch.Tensor) ->
         dist: Distortion coefficients
             :math:`(k_1,k_2,p_1,p_2[,k_3[,k_4,k_5,k_6[,s_1,s_2,s_3,s_4[,\tau_x,\tau_y]]]])`. This is
             a vector with 4, 5, 8, 12 or 14 elements with shape :math:`(*, n)`.
+        new_K: New intrinsic camera matrix with shape :math:`(*, 3, 3)`. Default: None, in this case K is used.
```
Comment (edgarriba) [inline kornia/geometry/calibration/distort.py:73]: Add more explanation like in the OpenCV function, maybe just rephrase that one. Btw, I see they use identity instead

File: kornia/geometry/calibration/distort.py (hunk 1 above)
Comment (lferraz) [reply kornia/geometry/calibration/distort.py:108]: used in line 148

File: kornia/geometry/calibration/undistort.py (hunk 2 above)
Comment (lferraz) [reply kornia/geometry/calibration/undistort.py:67]: used in line 107

File: kornia/geometry/calibration/distort.py (hunk 4 above)
Comment (lferraz) [reply kornia/geometry/calibration/distort.py:73]: I will try but explanation in opencv is ultra bad. They are doing the same as in this code.

File: kornia/geometry/calibration/distort.py (hunk 1 above)
Comment (lferraz) [reply kornia/geometry/calibration/distort.py:108]: moved def of vars to the place where they are used

File: kornia/geometry/calibration/undistort.py (hunk 2 above)
Comment (lferraz) [reply kornia/geometry/calibration/undistort.py:67]: moved def of vars to the place where they are used

File: kornia/geometry/calibration/undistort.py (hunk 3 above)
Comment (lferraz) [reply kornia/geometry/calibration/undistort.py:28]: done here. What do you refer with "others"?

File: kornia/geometry/calibration/distort.py (hunk 4 above)
Comment (lferraz) [reply kornia/geometry/calibration/distort.py:73]: I found this description, I think is the best one: "Camera matrix of the distorted image. By default, it is the same as cameraMatrix but you may additionally scale and shift the result by using a different matrix."

**Review by edgarriba — APPROVED** [review]

**Comment by edgarriba** [comment]
@shijianjian can you check why some unrelated tests are broken?

**Comment by edgarriba** [comment]
@lferraz i suggest to merge and we later fix the issues in master ?

**Comment by edgarriba** [comment]
@lferraz please rebase

**Pull request merged**